### PR TITLE
first pass of asyncio in distributor

### PR DESCRIPTION
**Pull request by davidshaw-uw**
As a comment, copilot was _very_ nice in developing this PR

### CODE REVIEW

File: jobmon_core/src/jobmon/distributor/distributor_service.py [hunk 1]
```diff
@@ -411,24 +415,77 @@ def log_task_instance_report_by_date(self) -> None:
...
+            max_attempts -= 1
+            await asyncio.sleep(wait_time)
+            wait_time *= 2
+
+        if max_attempts == 0 and not http_request_ok(return_code):
             raise InvalidResponse(
-                f"{app_route} Returned={return_code}. Message={message}"
+                f"Unexpected status code {return_code} from POST "
+                f"request through route {app_route}. Expected "
+                f"code 200. Response content: {response}"
```
Comment (LimingXuIHME) [inline jobmon_core/src/jobmon/distributor/distributor_service.py:487]: Kinda think the "expected code 200" is uncessary

**Review by LimingXuIHME — APPROVED** [review]

File: jobmon_core/src/jobmon/distributor/distributor_service.py (hunk 1 above)
Comment (davidshaw-uw) [reply jobmon_core/src/jobmon/distributor/distributor_service.py:487]: Sure, that's just the default invaliderror raised elsewhere in the codebase.

File: jobmon_core/src/jobmon/distributor/distributor_service.py [hunk 2]
```diff
@@ -411,24 +415,77 @@ def log_task_instance_report_by_date(self) -> None:
...
-            app_route="/task_instance/log_report_by/batch",
-            message=message,
-            request_type="post",
-        )
-        if http_request_ok(return_code) is False:
+
+        # Super basic retrying logic, to avoid fussing with tenacity logic.
+        # TODO: Factor out into an asynchronous requester
+
+        max_attempts, wait_time = 10, 1.5
```
Comment (everdyke) [inline jobmon_core/src/jobmon/distributor/distributor_service.py:455]: Wonder if we should move these values somewhere where they're easier to change

**Review by everdyke — APPROVED** [review]

File: jobmon_core/src/jobmon/distributor/distributor_service.py (hunk 2 above)
Comment (davidshaw-uw) [reply jobmon_core/src/jobmon/distributor/distributor_service.py:455]: As per the todo, eventually we should adopt an asynchronous requester that implements its own retrying logic. For now, this is a custom implementation of exponential backoff with 10 max retries

Comment (everdyke) [reply jobmon_core/src/jobmon/distributor/distributor_service.py:455]: Somehow missed that todo lol

**Pull request merged**